include/hip/hcc_detail/helpers.hpp

```diff
@@ -48,7 +48,24 @@ namespace hip_impl {
 template <typename...>
 using void_t_ = void;
 
-#if (__cplusplus < 201402L)
+#if ((__cplusplus >= 201703L) && defined(__cpp_lib_is_invocable))
```

Comment:
This needs to include the `<version>` header for `__cpp_lib_is_invocable` macro and it needs to check the value(see [here](https://en.cppreference.com/w/cpp/feature_test#Library_features) on documentation). 

```cpp
#ifndef HIP_HAS_CPP17_INVOCABLE             // Check if the macro has already been defined, perhaps by the user
#  ifdef __has_include                      // Check if __has_include is present
#    if __has_include(<version>)            // Check for version header
#      include <version>
#      if __cpp_lib_is_invocable            // Check the value of __cpp_lib_is_invocable
#         define HIP_HAS_CPP17_INVOCABLE 1
#      else                            
#         define HIP_HAS_CPP17_INVOCABLE 0
#      endif
#    else                                   // Not found at all
#       define HIP_HAS_CPP17_INVOCABLE 0
#    endif
#  endif
#endif
```